csharp/src/Ice/LocatorInfo.cs

```diff
@@ -90,83 +90,82 @@ internal void ClearCache(Reference reference)
                 {
-                    if (_background && endpoints.Count > 0)
+                    try
                     {
-                        // Endpoints are returned from the cache but TTL was reached, if backgrounds updates
-                        // are configured, we obtain new endpoints but continue using the stale endpoints to
-                        // not block the caller.
-                        _ = ResolveLocationAsync(location,
-                                                 reference.Protocol,
-                                                 reference.Communicator,
-                                                 cancel: default);
+                        endpoints = await ResolveLocationAsync(location,
+                                                               reference.Protocol,
+                                                               reference.Communicator,
+                                                               cancel).ConfigureAwait(false);
+                        endpointsAge = TimeSpan.Zero; // Not cached
                     }
-                    else
+                    finally
                     {
-                        try
-                        {
-                            endpoints = await ResolveLocationAsync(location,
-                                                                   reference.Protocol,
-                                                                   reference.Communicator,
-                                                                   cancel).ConfigureAwait(false);
-                        }
-                        finally
+                        // If we can't resolve the location we clear the resolved well-known proxy from the cache.
+                        if (endpoints.Count == 0 && reference.IsWellKnown)
                         {
-                            // If we can't resolve the location we clear the resolved well-known proxy from the cache.
-                            if (endpoints.Count == 0 && reference.IsWellKnown)
-                            {
-                                ClearCache(reference);
-                            }
+                            ClearCache(reference);
                         }
                     }
                 }
+                else if (expired && _background)
+                {
+                    // Endpoints are returned from the cache but TTL was reached, if backgrounds updates
+                    // are configured, we obtain new endpoints but continue using the stale endpoints to
+                    // not block the caller.
+                    _ = ResolveLocationAsync(location, reference.Protocol, reference.Communicator, cancel: default);
+                }
 
                 if (!reference.IsWellKnown)
                 {
-                    cached = cachedLocation;
+                    locationAge = endpointsAge;
```

Comment:
If this `if` still needed? Can't we just always return `endpointsAge` instead of `locationAge`?